json.js

```diff
@@ -23,3 +12,14 @@ function replacer(key, value) {
     ? value.toString('base64')
     : value;
 }
+
+/*
+ * function json (info)
+ * Returns a new instance of the JSON format that turns a log `info`
+ * object into pure JSON. This was previously exposed as { json: true }
+ * to transports in `winston < 3.0.0`.
+ */
+module.exports = format((info, opts = { replacer, space: 0 }) => {
```

Comment:
Doing default assignments in a `format` is a non-trivial performance hit. Instead of defaulting to a fallback value we're creating a new object for every single log message.

Reply:
My previous statement was incorrect, but this is still not equivalent because `opts` is **guarenteed** to always be _at least_ an empty object (it is defaulted when the `Format` is created, and _reused_ on every log message). As such this default assignment never happens.